### ISSUE TITLE
Add1 and Add2 page reworked (account based)

### ISSUE THREAD

**Issue by lelmarir** [aegif/CmisSync#587]
As anticipated on #567 i'm thinking to rewrite the Add1 and Add2 page (server/credentials selection and folder selection).

*For Add1*
The idea is to implement a dropdown or a list in which the user can manage accounts (server url with username and password) and, chosing one, create multiple sync folder on the same account.
This way whenever any data change (url, username or password), updating the account will get all synced folders working again. (and the user don't need to enter again and again the same data)

*For Add2*
Here more or less a refactor, using the [ViewModel Pattern for the WPF TreeView](http://www.codeproject.com/Articles/26288/Simplifying-the-WPF-TreeView-by-Using-the-ViewMode).

I'd like also to add checkboxes to repository/folder list. The idea is that the user can check which folder he need to syncronize (only one root) and uncheck subfolders to exclude them. But still an early idea.

**Comment by nicolas-raoul** [comment]
About Add1:
Would this be transparent for the config.xml file? Or would you introduce a new "credentials" node whose subnodes are references from the "folder" nodes?

About the checkboxes idea:
- It is important that local path don't become too deep. For instance, let's say I want to sync project1 and project2, both of which are in the same site. `C:\Users\nico\CmisSync\project1` and `C:\Users\nico\CmisSync\project3` are perfect but `C:\Users\nico\CmisSync\Sites\myteam\documentLibrary\2015\project1` and `C:\Users\nico\CmisSync\Sites\myteam\documentLibrary\2015\project3` would be too long, users would give using such deep paths.
- It is also important that people don't get folders they did not subscribe for. For instance, let's say I synchronized only project1 and project3, and later some coworker creates project46. Then project46 should not be synchronized to my local computer. In other words, selection should be *inclusive* (include only project1 and project3) not *exclusive* (exclude project2).

Is my explanation understandable?
Maybe we could do a Skype or Hangout some day? :-) Email me at nicolas.raoul at aegif.jp for my Skype number if you want. Cheers!

**Comment by lelmarir** [comment]
> Would this be transparent for the config.xml file? Or would you introduce a new "credentials" node whose subnodes are references from the "folder" nodes?

no, it is not, indeed i'll need to create a new node (i'll try to keep it retro compatible or handle a migration process)

> let's say I synchronized only project1 and project3
No, I badly explained myself.
The user can't chose project1 and project3. 

- He can chose the parent (Site) and exclude project2 (but as you said he will get project4). But in this case the synced folder would be Site (C:\Users\nico\CmisSync\Sites)
- can chose project2, adding another folder (repeating the procedure) chose project3. In this case the folders would be C:\Users\nico\CmisSync\project2 and C:\Users\nico\CmisSync\project3.

The purpose of checkboxes is to exclude subfolders, not to sync multiple ones.
Is this passable?

> Skype or Hangout 

I'm not so good at spoken english, but we can try if you like. I'll send you an email.
